backend/src/main/java/de/adrianaschepers/shishabuddies/service/UserService.java

```diff
@@ -55,7 +55,11 @@ public UserEntity createUser(UserEntity userEntity) {
     public SettingsEntity getUserSettings(UserEntity authUser) {
         Optional<UserEntity> userOpt = userRepository.findByUserName(authUser.getUserName());
         if(userOpt.isPresent()) {
-            return userOpt.get().getSettings();
+            Optional<SettingsEntity> settingsOfAuth = Optional.ofNullable(userOpt.get().getSettings());
+            if(settingsOfAuth.isPresent()){
+               return settingsOfAuth.get();
+            }
```

Comment:
Why do you construct the second optional?

```suggestion
            SettinfsEntity entity = userOpt.get().getSettings();
            if(entity != null){
              return entitis;
            }
```

Imho, You gain very little by constructing the optional to use it directly after.